nservicebus/upgrades/7to8/message-contracts.md

```diff
@@ -0,0 +1,47 @@
+---
+title: Upgrading message contracts from Version 7 to 8
+summary: Instructions on how to upgrade shared message contracts from NServiceBus version 7 to version 8.
+reviewed: 2022-02-02
+component: Core
+isUpgradeGuide: true
+upgradeGuideCoreVersions:
+ - 7
+ - 8
+---
+
+NServiceBus version 8 does not support .NET Standard as a framework target. Therefore, message contract assemblies that target `netstandard2.0` and that reference NServiceBus must be updated. This guidance outlines possible approaches.
+
+Note: When using [unobtrusive mode](https://docs.particular.net/nservicebus/messaging/unobtrusive-mode), the contracts assembly doesn't require a reference to NServiceBus and therefore is not affected. Unobtrusive message contracts can continue to target .NET Standard.
+
+## Change to specific target Platform
+
+If all endpoints are targeting the same platform (e.g. .NET Core 3.1), the message contracts assembly can be changed to align with all other endpoint projects.
+
+## Multi-targeting
+
+If endpoints that share a common contracts assembly target different platforms and frameworks (e.g. both .NET Framework 4.8 and .NET Core 3.1), the target assembly can use [multi-targeting](https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/cross-platform-targeting#multi-targeting) by replacing the `TargetFramwork` element with the `TargetFrameworks` element in the `.csproj` settings:
```

Comment:
```suggestion
If endpoints that share a common contracts assembly target different platforms and frameworks (e.g. both .NET Framework 4.8 and .NET Core 3.1), the target assembly can use [multi-targeting](https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/cross-platform-targeting#multi-targeting) by replacing the `TargetFramwork` element with the `TargetFrameworks` (note the plural) element in the `.csproj` settings:
```

Not sure it's worth it. I always miss that, though.

Reply:
that's why I added both elements in the text, but I guess it wouldn't hurt to mention this even more explicitly.